### DISCUSSION TITLE
Are multiple compute pipelines parallel or serial？

### DISCUSSION THREAD

**Discussion by juzi5230** [gfx-rs/wgpu#2999] [Q&A]
As the title, in the webgpu, I create two compute pipelines to compute different attributes, I wonder the compute pipelines is parallel or serial.

If they are serial, whether I can change them to be parallel.

thank you！

The following is the code: 

    const commandEncoder = device.createCommandEncoder()
    const computePass = commandEncoder.beginComputePass()
    computePass.setPipeline(pipelineObj.computePipeline)
    computePass.setBindGroup(0, pipelineObj.computeGroup)
    computePass.dispatchWorkgroups(Math.ceil(NUM / 128))
    computePass.end()

    const computePass1 = commandEncoder.beginComputePass()
    computePass1.setPipeline(pipelineObj.computeLightPipeline)
    computePass1.setBindGroup(0, pipelineObj.lightGroup)
    computePass1.dispatchWorkgroups(Math.ceil(NUM / 128))
    computePass1.end()

**Comment by teoxoy** [comment]
As long as the resources used and passed to the shaders differ (i.e. if the pipelines don't depend on each other), they will run in parallel.

**Reply by cwfitzgerald** [reply]
I should note, they are _allowed_ to run in parallel, if they do is up to the implementation, driver, and gpu capabilities.
